src/Sulu/Bundle/CoreBundle/Resources/config/rest.xml

```diff
@@ -21,10 +21,63 @@
                  class="Sulu\Component\Rest\ListBuilder\Doctrine\DoctrineListBuilderFactory"
                  public="true">
             <argument type="service" id="doctrine.orm.entity_manager"/>
+            <argument type="service" id="sulu_core.list_builder_filter_type_registry"/>
             <argument type="service" id="event_dispatcher"/>
             <argument>%sulu_security.permissions%</argument>
         </service>
-        <service id="Sulu\Component\Rest\ListBuilder\Doctrine\DoctrineListBuilderFactoryInterface" alias="sulu_core.doctrine_list_builder_factory"/>
+        <service
+            id="Sulu\Component\Rest\ListBuilder\Doctrine\DoctrineListBuilderFactoryInterface"
+            alias="sulu_core.doctrine_list_builder_factory"
+        />
+
+        <service
+            id="sulu_core.list_builder_filter_type_registry"
+            class="Sulu\Component\Rest\ListBuilder\Filter\FilterTypeRegistry"
+        >
+            <argument type="tagged" tag="sulu_core.list_builder_filter_type" index-by="alias" />
```

Comment:
would prefer if we would start using `default_index_method` with a static method on the class. this will make things easier for the developer because he can just autowiring in a project

Reply:
Would do it, if I could get it working :see_no_evil: Should it work if I just omit the `index-by` definition here, and add a `public static function getDefaultIndexName` to the `FilterType`s?

Reply:
no, you need to define the `default_index_method` as attribute of this argument 🙂 see last secion of https://symfony.com/blog/new-in-symfony-4-3-indexed-and-tagged-service-collections

Reply:
If I am reading the article correctly, this attribute is optional, and only necessary if you want to use something different than `getDefaultIndexName`. However, even when I've tried to add this, the XSD schema complained about it, because this attribute does not seem to exist there :confused:

Reply:
Although, maybe I was using underscores instead of dashes... gonna try it again.

Reply:
okay looks like you still ned a `index-by` attribute: https://symfony.com/doc/4.4/service_container/service_subscribers_locators.html#indexing-the-collection-of-services

Reply:
Okay, the problem was something different... This only works, if you keep the `index-by` attribute as well...